src/analysis/vector/geometry_checker/qgsgeometrymissingvertexcheck.h

```diff
@@ -34,12 +34,15 @@ class QgsCurvePolygon;
  */
 class ANALYSIS_EXPORT QgsGeometryMissingVertexCheck : public QgsGeometryCheck
 {
+    Q_GADGET
+
   public:
     enum ResolutionMethod
     {
       NoChange,
       AddMissingVertex
     };
+    Q_ENUM( ResolutionMethod )
```

Comment:
just thinking we could use a template for collect/fix errors that would take the resolution method enums?

Reply:
I guess the code for Q_GADGET, Q_ENUM etc. cannot be put into a template, and a template with only half the code is not soooo nice.